Resolve selenium.common.exceptions.WebDriverException errors

deanmalmgren opened the pull request:
This is almost certainly due to a compatibility between the version of `selenium` and  Firefox. We've seen a couple of errors from this module:

```
selenium.common.exceptions.WebDriverException: Message: 'wires' executable needs to be                 available in the path.
``` 

```
selenium.common.exceptions.WebDriverException: Message: The browser appears to have exited before we could connect. If you specified a log_file in the FirefoxBinary constructor, check it for details.
```

michaelmoliterno commented:
@deanmalmgren I think I made some progress on this. The steps below got me almost to functionality (will explain almost below):

Environment variable update (unsure if this is actually required):
```
export DISPLAY=:10.0
```

Package upgrades:

selenium==2.48.0 > selenium==3.3.1
pyOpenSSL==0.15.1 > pyOpenSSL==16.2.0

```
pip install -U selenium
pip install -U pyOpenSSL
```

If you try that, I bet it works. When I run `sync_quickbooks_gdrive.py`, Firefox starts and begins to authenticate with Quickbooks (i.e. fills in the user/pw, submit) but then runs into the screen below:

<img width="906" alt="screen shot 2017-03-28 at 10 27 59 pm" src="https://cloud.githubusercontent.com/assets/9665491/24437329/076b6124-1406-11e7-8933-db3984c4faba.png">

I think that because I've never logged into Quickbooks with Firefox, it's doing this (I remember needing to do this for Chrome too). I think I need to request that one-time code and then should be good to validate that this is working (I hope) -- let's try Wednesday?

deanmalmgren commented:
@michaelmoliterno I converted your issue into a pull request and I added your changes to the `requirements/python` on this `selenium-fix` branch. Now I'm running into a problem that I believe we fixed on your computer:

```
Deep-Thought:  ~/Doodies/a-model  |selenium-fix {3} ?:3 ✗|
{a-model} $ sync_quickbooks_gdrive.py 
Traceback (most recent call last):
  File "/Users/rdm/Doodies/a-model/bin/sync_quickbooks_gdrive.py", line 8, in <module>
    from a_model import reports
  File "/Users/rdm/Doodies/a-model/a_model/reports/__init__.py", line 6, in <module>
    from . import base
  File "/Users/rdm/Doodies/a-model/a_model/reports/base.py", line 19, in <module>
    from oauth2client.client import SignedJwtAssertionCredentials
ImportError: cannot import name SignedJwtAssertionCredentials
```

Can you merge your changes into `selenium-fix` and we can try and get this resolved together?

michaelmoliterno commented:
Code we fussed around with that may or may not be related to this issue. 

```
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities

# http://stackoverflow.com/a/39324230
capabilities = DesiredCapabilities.FIREFOX
capabilities["marionette"] = True

kwargs.update({
'firefox_profile': profile,
'capabilities': capabilities,
 })
```
